.github/actions/tests/add-zone-query/action.yml

```
@@ -0,0 +1,44 @@
+# Making reusable composite actions documented at
+# https://docs.github.com/en/actions/tutorials/create-actions/create-a-composite-action#creating-a-composite-action-within-the-same-repository
+name: 'Add a zone, approve the unsigned zone, query the published zone'
+description: 'Add a zone, approve the unsigned zone, query the published zone'
+defaults:
+  # see: https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-syntax#defaultsrunshell
+  shell: bash --noprofile --norc -eo pipefail -x {0}
+runs:
+  using: "composite"
+  steps:
+    - name: Prepare the system test environment
+      uses: ./.github/actions/prepare-systest-env
+      with:
+        artifact-name: ${{ format('cascade_{0}_{1}_{2}', github.sha, matrix.os, matrix.rust) }}
```

The composite action does not have access to the workflow's `matrix` variable:
```
::error::Unable to download artifact(s): Artifact not found for name: cascade_d2d664fa2e5dda4edae9b08922f89d06d08d7be5__%0A
```

```suggestion
        artifact-name: ${{ inputs.artifact-name }}
```